nbgitpuller/handlers.py

```diff
@@ -87,33 +153,22 @@ def pull():
                     q.put_nowait(e)
                     raise e
             self.gp_thread = threading.Thread(target=pull)
-
             self.gp_thread.start()
-
-            while True:
-                try:
-                    progress = q.get_nowait()
-                except Empty:
-                    yield gen.sleep(0.5)
-                    continue
-                if progress is None:
-                    break
-                if isinstance(progress, Exception):
-                    self.emit({
-                        'phase': 'error',
-                        'message': str(progress),
-                        'output': '\n'.join([
-                            line.strip()
-                            for line in traceback.format_exception(
-                                type(progress), progress, progress.__traceback__
-                            )
-                        ])
-                    })
-                    return
-
-                self.emit({'output': progress, 'phase': 'syncing'})
-
+            self.progress_loop(q)
```

Comment:
Is it correct to not await this? I figure, this is a coroutine or async function, then we should await/yield it right?

Reply:
I  added yield here OK; so by not yielding were blocking on that line across the server. I am still wrapping my mind around these async pieces.

Reply:
my OK here was misplaced! I meant "I add yield here; OK so, ..." hope that did not come off as offensive!

Reply:
I think by not yield (aka. await with modern syntax, also with modern syntax the gen.coroutine decorated functions map to async prefixed functions), this would just return a future instantly and then the next line of code would execute. So, it would make the function complete fast, but with things left to do separately. Now, the function calling progress loop will not return until progress loop returns.

Reply:
OK yes. This is making sense to me. Good deal